doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1023 @@
+
+Description: Updated existing table to store VNI associated with VRF.
+
+Schema:
+
+```
+;Existing table
+;defines virtual routing forward table. Updated to stores VNI associated with VRF
+;
+;Status: stable
+
+key = VRF_TABLE:VRF_NAME ;
+fallback = "true"/"false"
+vni = 1*8DIGIT ; VNI associated with VRF
+
+```
+
+### 4.2.2 APP_DB Changes
+
+**EVPN_NVO_TABLE**
+
+Producer:  VxlanMgr 
+
+Consumer: VxlanOrch
+
+Description: This is a singleton configuration which holds the VTEP source IP used for BGP-EVPN discovered tunnels. 
+
+Schema:
+
+```
+; New table
+; holds the VTEP source IP used for BGP-EVPN discovered tunnels
+
+key = EVPN_NVO_TABLE:nvo_name
+                          ; nvo or VTEP name as a string
+; field = value
+source_vtep = tunnel_name ; refers to the VXLAN_TUNNEL object created with SIP only.
+```
+
+**EVPN_REMOTE_VNI** 
+
+Producer: fdbsyncd
+
+Consumer: VxlanOrch
+
+Description: This specifies the remote VNI to be extended over a tunnel. This is populated by fdbsyncd on receipt of a IMET route from the remote side. 
+
+Schema:
+
+```
+; New table
+; specifies the VLAN to be extended over a tunnel
+
+key = EVPN_REMOTE_VNI_TABLE:"Vlan"vlanid:remote_vtep_ip
+                  ; Vlan ID and Remote VTEP IP 
+; field = value
+vni = 1*8DIGIT    ; vni to be used for this VLAN when sending traffic to the remote VTEP
+```
+
+**VXLAN_FDB_TABLE**
+
+Producer: fdbsyncd
+
+Consumer: fdborch
+
+Description: This holds the MACs learnt over a VNI from the remote VTEP.  This is populated by fdbsyncd on receipt of a MAC route. 
+
+Schema:
+
+```
+; New table
+; specifies the MAC and VLAN to be programmed over a tunnel
+
+key = VXLAN_FDB_TABLE:"Vlan"vlanid:mac_address
+                          ;MAC Address and VLAN ID
+;field = value
+remote_vtep = ipv4
+type          = "static" / "dynamic"  ; type of the entry.
+vni         = 1*8DIGIT                ; vni to be used for this VLAN when sending traffic to the remote VTEP
+```
+
+**VRF_TABLE**
+
+Producer: VrfMgr
+
+Consumer: Vrforch
+
+Description: Updated existing table to store VNI associated with VRF.
+
+Schema:
+
+```
+;Existing table
+;defines virtual routing forward table. Updated to stores VNI associated with VRF
+;
+;Status: stable
+
+key = VRF_TABLE:VRF_NAME ;
+fallback = "true"/"false"             ; default false
+vni = 1*8DIGIT ; VNI assicated with the VRF
+
+```
+
+**VRF_ROUTE_TABLE**
+
+Producer: fpmsyncd
+
+Consumer: routeorch
+
+Description: VRF Route Table is extended for Tunnel Nexthop. 
+
+Schema:
+
+```
+;Stores a list of routes
+;Status: Mandatory
+
+key = ROUTE_TABLE:VRF_NAME:prefix ;
+nexthop = prefix,              ; IP addresses separated ',' (empty indicates no gateway). May indicate VXLAN endpoint if vni_label is non zero.
+intf = ifindex? PORT_TABLE.key ; zero or more separated by ',' (zero indicates no interface)
+vni_label = VRF.vni            ; New Field. zero or more separated by ',' (empty value for non-vxlan next-hops). May carry MPLS label in future.
```

Comment:
New fields added are part of Nexthop tuple {Nexthop IP, VNI, RMAC}. In case of Overlay ECMP, we will have Overlay Nexthops seperated by ',' . Nexthop IP and RMAC will be different. Encap VNI can be same or different.